### DISCUSSION TITLE
Unix Time format don't working

### DISCUSSION THREAD

**Discussion by trialsin** [thunderclient/thunder-client-support#571] [General]
Hi,

I'm trying format date using {{#date , 'X'}} to get Unix Time in seconds, but don't works. Here is documentation of moment:

https://momentjs.com/docs/#/displaying/format/

![image](https://user-images.githubusercontent.com/35371595/165386474-2383a5fe-70f3-40d7-90df-c2817adac82b.png)

**Comment by rangav** [comment]
Hi @trialsin we are not using momentjs library, so `X` format will not work 

only below supported 
date format:  `{{#date, 'YYYY-MM-DD hh:mm:ss:fff'}}`

Docs
https://github.com/rangav/thunder-client-support#variables

**Reply by trialsin** [reply]
Oh! I see!

there are some other way to create a unix timestamp or some way to make some simple maths?

Thank you!

Congratulations for your great extension

**Reply by rangav** [reply]
Just use the date `{{#date}}` it will generate timestamp

**Reply by trialsin** [reply]
Cool! But it seems be in miliseconds.

I need it in seconds and some minutes ahead

**Reply by rangav** [reply]
This feature will be part of next update

**Reply by trialsin** [reply]
Wow! Great!

**Comment by rangav** [comment]
new feature request created for this https://github.com/rangav/thunder-client-support/issues/599

**Comment by rangav** [comment]
This feature is now implemented, please update to version `v1.16.4`

See all features and bugs fixed here
https://github.com/rangav/thunder-client-support/releases/tag/v1.16.4

Docs:
https://github.com/rangav/thunder-client-support#variables

**Reply by trialsin** [reply]
Hello,

There are a little bug in validation when trying to send the timestamp as number:

![image](https://user-images.githubusercontent.com/35371595/171696086-bf95f1f7-3f9b-45cb-94c4-94a6fcad83cd.png)

And another bug in hotkey Ctrl+Enter to send the request: if I click in another text box and click again in Json Content, the hotkey don't works. It seems works again after I've made some editing in Content.

**Reply by rangav** [reply]
@trialsin its only complaining that it’s not valid json because of variables, if you send request it should work.

For formatting to work variables need custom logic which is in roadmap.

I will verify the Ctrl + Enter issue.
